test/e2e/bicg/bicg_kernel.mlir

```diff
@@ -0,0 +1,65 @@
+// Compile the C kernel to LLVM IR (let clang handle headers and macros).
+// Use -I %S so local headers (bicg.h, polybench.h) are visible.
+// RUN: clang -S -emit-llvm -O3 -fno-vectorize -fno-unroll-loops -std=c11 \
+// RUN:   -I %S/../../benchmark/CGRA-Bench/kernels/bicg -DSMALL_DATASET \
+// RUN:   -o %t-kernel-full.ll %S/../../benchmark/CGRA-Bench/kernels/bicg/bicg.c
+
+// Extract only the kernel function(s). PolyBench typically uses kernel_bicg,
+// so a regex keeps this robust across name variants.
+// RUN: llvm-extract --rfunc=".*kernel.*" %t-kernel-full.ll -o %t-kernel-only.ll
+
+// Import the LLVM IR into MLIR (LLVM dialect).
+// RUN: mlir-translate --import-llvm %t-kernel-only.ll -o %t-kernel.mlir
+
+// Lower and map to the Neura accelerator, then generate code.
+// Exact mapping (tiles, II, etc.) depends on the architecture/heuristics,
+// so checks below focus on structural properties for stability.
+// RUN: mlir-neura-opt %t-kernel.mlir \
+// RUN:   --assign-accelerator \
+// RUN:   --lower-llvm-to-neura \
+// RUN:   --promote-func-arg-to-const \
+// RUN:   --fold-constant \
+// RUN:   --canonicalize-live-in \
+// RUN:   --leverage-predicated-value \
+// RUN:   --transform-ctrl-to-data-flow \
+// RUN:   --fold-constant \
+// RUN:   --insert-data-mov \
+// RUN:   --map-to-accelerator="mapping-strategy=heuristic" \
+// RUN:   --architecture-spec=%S/../../arch_spec/architecture.yaml \
+// RUN:   --generate-code -o %t-mapping.mlir
+
```

Comment:
Can you show/`CHECK` the entire IR before `--canonicalize-live-in`? so we can see what const is not folded? (it is okay to not fold both const operand for a binary operation, as most operations in MLIR/LLVM requires at least one operand. Instead, we can have such RTL const operation in HW.)